web/ui/react-app/src/pages/alerts/CollapsibleAlertPanel.tsx

```diff
@@ -25,27 +25,10 @@ const CollapsibleAlertPanel: FC<CollapsibleAlertPanelProps> = ({ rule, showAnnot
     <>
       <Alert fade={false} onClick={() => toggle(!open)} color={alertColors[rule.state]} style={{ cursor: 'pointer' }}>
         <FontAwesomeIcon icon={open ? faChevronDown : faChevronRight} fixedWidth />
-        <strong>{rule.name}</strong> ({`${rule.alerts.length} active`})
+        <strong>{replaceWith(rule.name, `"`, `'`)}</strong> ({rule.alerts.length} active)
```

Comment:
This seems incorrect, since it would display an alert with the name `foo"bar` as having the name `foo'bar`.

I think in this place here you don't actually need to escape or change anything about the name. But in places where the name appears in double quotes (like in the graph link), you will need to escape double quotes in the name `"` -> `\"`.

Reply:
I was playing with that yesterday and found that query like ```foo"bar``` wrapped in quotes will not work since it became ```"foo"bar"```
Adding an escaping slash also will not work since it just prevents the error - the final result also will be ```"foo"bar"``` which causes the response from the query to be an error as well which isn't correct wright? It should return ```Empty query result```

Reply:
Hmm, not sure what you mean. The final query should become `ALERTS{alertname="foo\"bar"}`, which correctly produces an empty result (because no alert with this name is currently actually firing or even existing).

http://demo.robustperception.io:9090/graph?g0.range_input=1h&g0.expr=ALERTS%7Balertname%3D%22foo%5C%22bar%22%7D&g0.tab=1

So just escaping quotes with a slash should be fine, when used as a label value in a selector surrounded by quotes?

Reply:
Seems I was trying something different. I will revisit it.

Reply:
Having to manually escape HTML is a smell, do he libraries we use not take care of this for us?

Reply:
This is not about escaping HTML, it's about escaping quotes in PromQL strings (which should be a simple `"` -> `\"`).

Reply:
If that's the case, why is this in a line that includes HTML tags?

Reply:
The actual query doesn't contain any HTML tags. Those you mention are part of the UI

Reply:
Yep that's why I said in this place there doesn't need to be any escaping (on the HTML or PromQL level at least). HTML escaping is already automatically being done by React. But we do need to escape quotes in label values when they become part of a PromQL string like in the `ALERTS{alertname="foo\"bar"}` example that I mentioned (the relevant part of the code for that is furtehr below).

We do, however, strictly still have a bug here when it comes to YAML escaping. E.g. when the alert name contains any characters that have special meaning in YAML, we would need to detect that and quote it somehow. But at the moment the YAML is not copy-and-pastable anyway. See https://github.com/prometheus/prometheus/pull/6708#issuecomment-580221755